src/rdkafka_aux.c

```diff
@@ -248,6 +248,54 @@ rd_kafka_Node_t *rd_kafka_Node_new(int32_t id,
         return ret;
 }
 
+/**
+ * @brief Create a new Node object given a node id, and use broker information
+ * to populate other fields.
+ *
+ * @return A new allocated Node object.
+ *         Use rd_kafka_Node_destroy() to free when done.
+ * @remark The \p brokers_internal array is asumed to be sorted by node id,
+ *         while the \p brokers array is not. This is the current default from
+ *         Metadata response parsing.
+ */
+rd_kafka_Node_t *rd_kafka_Node_new_from_brokers(
```

Comment:
The only reservation I have about this is, about the assertion

```
        /* We assume that the marshalled representation is
         * no more than 4 times larger than the wire representation.
```

Since we're allocating the tmpabuf using this. 
I'm making the change by adding the sorted one, let me know if you want me to change the factor 4